docs/cluster_api.md

```diff
@@ -93,3 +80,37 @@ In more detail:
   - LastTransitionTime [string]
   - Message [string]
   - Reason [string]
+
+### MachineSetConfig
+  - `NodeType [NodeType:string]` *Enumerated node type for the machine set (compute or master)*
+  - `Infra [boolean]` *True if this machine set should contain infrastructure pods (registry, router)*
+  - `Size [int]` *How many machines should be created for this machine set*
+  - `Version [corev1.LocalObjectReference]` *Reference to ClusterVersion resource*
+  - `Hardware`
+    - `AWS` *-optional- contains AWS specific configuration*
+      - `InstanceType [string]` *EC2 instance type (ie. m4.xlarge)*
+  - `NodeLabels [map[string]string]` *Labels to apply to nodes created in this MachineSet*
+
+### ClusterVersion
+Standalone resource that encapsulates details of a named version. It
+defines general OpenShift details about a particular version such as 
+the image template and RPM version as well as cloud-specific information
+such as the AWS AMI name.
+
+A set of predefined ClusterVersion resources can be initially created in a shared 
+namespace (like `openshift`) to define the versions supported out of the box.
+Users are able to create their own in their own namespace and associate their
+clusters with that.
+
+- `ObjectMetadata (inline) [metav1.ObjectMetadata]`
+- `TypeMetadata (inline) [metav1.TypeMetadata]`
+
+#### Spec
+- `ImageURL [string]` *-optional- template for images to be used on the cluster (oreg_url in ansible)*
+- `ImageTag [string]` *-optional- the image tag for OpenShift images (either ImageURL or ImageTag can be specified, but not both, openshift_image_tag in ansible)*
+- `Release [string]` *-optional- simple release name (openshift_release in ansible)*
+- `RPMRepository [string]` *-optional- URL for the RPM repository to use*
```

Comment:
An array might be useful here. Also we might want a struct for an RPM repository to cover those basic settings we need. (name, URL, gpg key, gpgcheck, etc)

Reply:
ack, the struct makes sense.
What would be the purpose of the array?

Reply:
Was thinking if we needed to land more than one repo on the system. (EPEL for example) Maybe that's just past experience and not revelant now, when I worked on cluster lifecycle my typical inventory had two repos, one for the OCP bits and one for RHEL nightly extras.